src/components/Movie/MovieDetails.js

```diff
@@ -7,61 +9,42 @@ import SimilarMovies from './SimilarMovies';
 import MovieBanner from './MovieBanner';
 import './moviedetails.css';
 
+import * as movieActions from '../../actions/movieActions';
+
 require('es6-promise').polyfill();
 require('isomorphic-fetch');
 
-export default class MovieDetails extends Component {
+export class MovieDetails extends Component {
     constructor(props) {
         super(props);
-        this.state = {
-            movie: {},
-            selectedGenre: '',
-            similarMovies: []
-        };
     }
     componentDidMount() {
         let url = `${Constants.baseURL}/${this.props.match.params.id}`;
-        this.getRequest(url).then(response => {
-            this.populateState(response);
-            this.fetchSimilarMovies(response.genres[0]);
-        });
-    }
-    componentWillReceiveProps(nextProps) {
-        let url = `${Constants.baseURL}/${nextProps.match.params.id}`;
-        this.getRequest(url).then(response => {
-            this.populateState(response);
-            this.fetchSimilarMovies(response.genres[0]);
-        });
-    }
-    populateState(response) {
-        this.setState({
-            movie: response
-        });
-    }
-    fetchSimilarMovies(genre) {
-        let url = `${Constants.baseURL}?search=${genre}&searchBy=genres&sortBy=release_date&sortOrder=desc`;
-        this.getRequest(url)
-            .then(response => {
-                this.populateStateForSimilarMovies(genre, response.data);
-            });
-    }
-    getRequest(url) {
-        return fetch(url).then(res => res.json());
-    }
-    populateStateForSimilarMovies(genre, response) {
-        this.setState({
-            selectedGenre: genre,
-            similarMovies: response
-        });
+        this.props.actions.fetchMovieDetails(url);
     }
     render() {
         return (
             <div>
                 <Header showSearchButton={true} />
-                <MovieBanner movie={this.state.movie} />
-                <SimilarMovies genre={this.state.selectedGenre} movies={this.state.similarMovies} />
+                <MovieBanner movie={this.props.movie} />
+                <SimilarMovies genre={this.props.selectedGenre} movies={this.props.similarMovies} />
                 <Footer />
             </div>
         );
     }
 }
+
+function mapStateToProps(state, ownProps) {
+    return {
+        movie: state.movie.movie,

```

Comment:
One point here, that is a must, but it's something that is good to you know is about Seletors.

The concept of selectors is basically one thing to not expose your state directly to you component, in that case you have freedom to update your state without force your containers to knowing that change.

```javascript
export const mapStateToProps = (state) => ({
    movie: getMovie(state), // internally it can be just the (state) => state.movie.movie
    selectedGenre: getSelectedGenre(state),
    similarMovies: getSimilarMovies(state), 
}); 
```

Reply:
Done. Moved all selectors to their own file under "selectors" subdirectory.